### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by Lavanyagaur22**
### Description

Added documentation for background offline capabilities of the library.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -104,6 +104,105 @@ Add the following dependency in your **module's build.gradle**.
...
+   - If the network is connected, then run a loop on list of mutations.
+   - On each mutation fetched from the list, create an object of Mutation<D,T,V> by passing the fetched mutation to the 
+     `getMutation(mutation)` function of library.
+   - You can use the above object while creating an instance of ApolloCall and make a call to the server.
+   - Make sure to delete the fetched mutation from the list in the onResponse() of your callback by calling              
+   `deleteMutation(mutation)`.
+   
+ - **In Kotlin**
+  
+  ```kotlin
```
Comment (harshithdwivedi) [inline README.md]: Show the entire worker class please

File: README.md [hunk 2]
```diff
@@ -104,6 +104,105 @@ Add the following dependency in your **module's build.gradle**.
...
+        /*
+         getListOfMutations() returns the list of mutations stored in database in the library.
+         It's present in the parent class, i.e OffixWorker.
+         */
+        val listOfMutations = getListOfMutations()
+
+        /*
+         Check again if the network connection is there or not.
+         */
+        if (Offline.isNetwork()) {
```
Comment (harshithdwivedi) [inline README.md]: The worker will be executed only if there's an active network.
Also, if you want to add a network check; do it inside the forEach loop.

Same applies for the sample app.

File: README.md [hunk 3]
```diff
@@ -123,11 +222,9 @@ Add the following dependency in your **module's build.gradle**.
 
 ## Limitations
 
-1. Offline mutations are lost when the app is closed.
-2. No UI Bindings, you will have to update your UI manually.
+1. No UI Bindings, you will have to update your UI manually.
+2. This library can only be used by integrating with Apollo Framework as of now.
```
Comment (harshithdwivedi) [inline README.md]: The library only supports apollo-android for the time being

**Review by harshithdwivedi — CHANGES_REQUESTED** [review]
Changes needed

**Comment by wtrocki** [comment]
I tend to provide examples for certain use cases in code. Do we have this implemented somewhere?

Also README becomes very large and technical. This is an antipattern for os projects. We might just restore docs folder and use something like docsify

**Comment by wtrocki** [comment]
README covers so many optional stuff etc. that is hard to wrap head. We should split this into getting started. Working in background etc.

**Comment by Lavanyagaur22** [comment]
> I tend to provide examples for certain use cases in code. Do we have this implemented somewhere?

Hey, Can you please explain it in a bit detail, like what use cases you are talking about?

**Comment by harshithdwivedi** [comment]
@Lavanyagaur22 maybe it's a good idea to update the docs according to the latest PR that was merged.
Also, I agree with @wtrocki 's suggestion; maybe we can branch out the docs to a wiki once we have the offline part tested.

**Comment by wtrocki** [comment]
> Hey, Can you please explain it in a bit detail, like what use cases you are talking about?

Screenshot 2019-07-23 at 14 37 15

File: README.md [hunk 4]
```diff
@@ -11,218 +11,14 @@
 Mutations are persisted when Offline
 and replicated back to server when online.
 
+## Documentation
 
+Link to be added.
```
Comment (wtrocki) [inline README.md]: I can register domain. android.offix.dev is ok?

File: README.md [hunk 5]
```diff
@@ -11,123 +11,25 @@
...
 
 ## Example application
 
 See `sample` for example application.<br/>
-For now it's using ionic showcase as its backened (https://github.com/aerogear/ionic-showcase)
-
-## Setup
```
Comment (wtrocki) [inline README.md:21]: Where this is going? Love to have it in docs folder.

File: README.md (hunk 4 above)
Comment (harshithdwivedi) [reply README.md]: Yep, perfect! ❤️

**Comment by Lavanyagaur22** [comment]
@harshithdwivedi  Please have a look at it.

**Comment by harshithdwivedi** [comment]
We need to have basic setup in readme as well.

For example, we can have : 
1. Overview
2. Installation 
3. Usage (link the docs here)

**Comment by harshithdwivedi** [comment]
Look at the docs for okhttp, etc. To get more idea

**Comment by Lavanyagaur22** [comment]
@harshithdwivedi Please have a look at it.

**Comment by Lavanyagaur22** [comment]
After you approve this readme, will copy paste this readme in project website readme (first page)

**Review by harshithdwivedi — APPROVED** [review]

**Pull request merged**